### ISSUE TITLE
PKP_Bdiff and PKP_Cdiff

### ISSUE THREAD

**Issue by letwave** [crotwell/TauP#3]
Dear Philip,

The message is just for reminder to add support for the two phases into Taup package in the future.

All the best,

Xiaobo

**Comment by crotwell** [comment]
Can you also provide a description of the paths these phases take so I don't forget.

**Comment by letwave** [comment]
1. PKP_Bdiff is turning within the middle of the outer core,  observed at epicentral distances closer to B-caustic point (~145 degree), and it can be generally observed in between 137.5 - 142.5 degree, and has and apparent slowness of ~3.8 sec/deg.
2. PKP_Cdiff is diffracting along the inner-core boundary, observed at epicentral distances larger than 155 degree (C- cut off point), and it can be generally observed in between 154.4 - 162.6 degree and has an apparent slowness of ~2.07 sec/deg.

**Comment by crotwell** [comment]
I feel TauP could be expanded to handle the inner core diffraction (PKP_Cdiff) as that path is comparable to Pdiff, just along a different boundary. However, I am not clear as to how the PKP_Bdiff phase, turning in the outer core, would be handled. Can you explain how this phase is different from plain old PKP? In particular, I don't see how you would generate a diffracted wave in the outer core without a corresponding discontinuity in the model in the outer core?

In other words, just because a phase exists in the earth and can be observed in real data doesn't mean that it can be computed with a 1D ray theoretic algorithm and within a particular model of the earth. And if the phase is computable, I need a very specific description of the path in order to code it.

**Comment by letwave** [comment]
At the distances of < ~145, there is no plain old PKP, however, as a result of finite frequency effect, waves diffracting into the shadow zone of outer core are still observable, and which are not predicable by ray-theory. Probably they (PKP_Bdiff) can be  roughly predicted by some approximation algorithm to approach the true solution of wave equation.

PKP_Cdiff can be handled as Pdiff has been in TauP package.

**Comment by crotwell** [comment]
I have added support for phases like PKdiffP, diffraction along inner-outer core boundary. Which I think addresses your PKP_Cdiff phase. 

Given PKP_Bdiff does not exist as a ray-theory arrival, I think it best to not try to include it.
